channeldb/payments_test.go

```diff
@@ -422,3 +423,63 @@ func TestQueryPayments(t *testing.T) {
 		})
 	}
 }
+
+// appendDuplicatePayment adds a duplicate payment to an existing payment. Note
+// that this function requires a unique sequence number.
+func appendDuplicatePayment(t *testing.T, db *DB, paymentHash lntypes.Hash,
```

Comment:
There is quite a bit of code in this PR to deal with duplicate payments. I am wondering for how long we are going to invest time in those? Maybe there is some kind of middle ground where we don't delete them completely, but only make them retrieval through a special endpoint. They would then also not need the indexing changes and all the associated test code

Reply:
I think it's worth dealing with legacy in a way that as as seamless to the user as possible, extra endpoints feels like another thing for people to load up into their mental model of how lnd works. If we want to sunset duplicates, I think we'd at least need to communicate that pretty widely, then perhaps have a release which deletes and csv-dumps those payments for the user. We could go down that road, but I think it's a long term plan, and people are having issues with their `ListPyayments` endpoint now, so I'd argue for tolerating them once more and then making a proper deprecation plan.